media-gfx/plantuml/plantuml-1.2024.4-r1.ebuild

```diff
@@ -0,0 +1,46 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+JAVA_PKG_IUSE="doc source"
+inherit java-pkg-2 java-pkg-simple desktop xdg
+
+DESCRIPTION="Draw UML diagrams using a simple and human readable text description"
+HOMEPAGE="https://plantuml.com"
+SRC_URI="https://github.com/${PN}/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/${P}"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND=">=virtual/jdk-1.8:*
+	>=dev-java/ant-1.10.14-r1:0"
+RDEPEND=">=virtual/jre-1.8:*
+	media-gfx/graphviz"
+
+JAVA_AUTOMATIC_MODULE_NAME="net.sourceforge.plantuml"
+JAVA_CLASSPATH_EXTRA="ant"
+JAVA_MAIN_CLASS="net.sourceforge.plantuml.Run"
+JAVA_RESOURCE_DIRS="res"
+JAVA_SRC_DIR="src"
+
+src_prepare() {
+	java-pkg-2_src_prepare
+	# It should also work without ant after removing these 2 files:
```

Comment:
this comment confuses me. you mean that the package should work without ant in the dependencies? and why is it mentioned here?

Reply:
> this comment confuses me. you mean that the package should work without ant in the dependencies? and why is it mentioned here?

If those two lines got removed, then also `ant` could be dropped from the dependencies, And `java-dep-check` would not complain. Haven't done any runtime tests, sorry.

Reply:
ah, thanks for explanation. i'm not sure we need to mention this. i guess plantuml has some feature where it can be used as an ant task and we should preserve that for users. i'll leave this up to you, either remove the comment or let me know that it's needed there and i would merge it as it is, maybe with some slight modification in the comment as it was not clear to me.

Reply:
> ah, thanks for explanation. i'm not sure we need to mention this. i guess plantuml has some feature where it can be used as an ant task and we should preserve that for users. i'll leave this up to you, either remove the comment or let me know that it's needed there and i would merge it as it is, maybe with some slight modification in the comment as it was not clear to me.

Comment removed.

Reply:
> i guess plantuml has some feature where it can be used as an ant task and we should preserve that for users.

Yes, it really looks like an [ant-task](https://github.com/gentoo/gentoo/commit/8a3c6f1678859118080a773eedb058903157b495#diff-7733e0ccb8fcd634736638bc9d0f247770ae3c812572ce22500a9b0a04e59d2eR32-R33). For the next release we could make it optional similar to [this proposal](https://github.com/gentoo/gentoo/pull/32820/commits/6e80ebc49fe4c59afa0ef9268e4cbaf41f071d58#diff-c1cc308d35d976bc6d4ef56fb1a62025f961696085ceaee129fce24dc8551023R19-R39). This would allow going without `ant` at the cost of some more lines in the `ebuild` plus a new USE flag.